routing/road_access_serialization.hpp

```diff
@@ -62,13 +62,14 @@ class RoadAccessSerializer final
   static void Deserialize(Source & src, VehicleType vehicleType, RoadAccess & roadAccess)
   {
     uint32_t const header = ReadPrimitiveFromSource<uint32_t>(src);
-    CHECK_EQUAL(header, kLatestVersion, ());
+    size_t const vehicleTypes = header >= 1 ? static_cast<size_t>(VehicleType::Count)
+                                            : 3 /* There were 3 vehicle types for version 0 */;
```

Comment:
The layout is also important.
I suggest splitting the code by version explicitly.

Reply:
Done.
I moved all version specific logic including order of subsections to `subsectionNumberToVehicleType`

Reply:
Well... it's a solution too. What I had in mind is separating the v0 code and never touch it afterwards, even if it may result in some code duplication.